File: SeongYeon/ts/1932.ts
// input
const input: string[] = require('fs')
  .readFileSync(process.platform === 'linux' ? '/dev/stdin' : 'input.txt')
  .toString()
  .trim()
  .split('\n');

// variable
const dp: number[][] = input.map((item) => item.split(' ').map(Number));
const n: number = dp[0]![0]!;

// solution
for (let i = 2; i <= n; i++) {
  for (let j = 0; j < i; j++) {
    if (j === 0) dp[i]![j] += dp[i - 1]![j]!;
    else if (j === i - 1) dp[i]![j] += dp[i - 1]![j - 1]!;
    else {
      dp[i]![j] += Math.max(dp[i - 1]![j]!, dp[i - 1]![j - 1]!);
    }
  }
}

// output
console.log(Math.max(...dp[n]!));

File: JiHwan/BJ/1932.py
N = int(input())

dp = []

for i in range(N):
    num = list(map(int, input().split()))
    dp.append(num)

for i in range(1, N):
    for j in range(0, i + 1):
        if j==0:
            dp[i][0] += dp[i-1][0]
        elif i==j:
            dp[i][j] += dp[i-1][j-1]
        else:
            dp[i][j] += max(dp[i - 1][j - 1], dp[i - 1][j])


print(max(dp[N - 1]))
